src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -1158,6 +1171,42 @@ async ValueTask LogDocumentChangesAsync(int? generation, CancellationToken cance
                         }
                     }
                 }
+
+                void UpdateChangedDocumentsStaleness(bool isStale)
```

Comment:
Since this local function closes over `changeOrAddedDocuments` and `changedDocumentStaleness`, which are defined 300 lines up and last referenced 200 hundred lines up, consider whether it makes sense to...

a) Move this local function closer to the declaration of these collections.
b) Make this local method static and pass both collections as arguments.
c) Introduce a helper struct to wrap the collections and this functionality.

Totally up to you. I just wanted to mention it since this method is 400+ lines long and this felt like it was a bit "spooky action at a distance".

Reply:
It'd be nice if C# allowed to infer local function types if I wanted to pass them as parameters:

```
void UpdateChangedDocumentsStaleness(var changedOrAddedDocuments, var changedDocumentsStaleness)
{
...
}
```